source/server/guarddog_impl.cc

```diff
@@ -102,7 +100,14 @@ void GuardDogImpl::step() {
                  static_cast<size_t>(ceil(multi_kill_fraction_ * watched_dogs_.size())));
 
     for (auto& watched_dog : watched_dogs_) {
-      const auto ltt = watched_dog->dog_->lastTouchTime();
+      if (watched_dog->dog_->touchCount() > 0) {
+        // Watchdog was touched since guarddog last checked; update time and reset the count.
+        watched_dog->dog_->resetTouchCount();
+        watched_dog->last_touch_time_ = now;
```

Comment:
So we'll lose some accuracy since the last touch time is set here instead of when the watchdog is actually touched. Is this ok? 

Should we then rename it to something else?

Reply:
It is still a timestamp from which the most recent touch was detected.  I agree that there's a change in precision; watchdog miss/megamiss/kill/multikill would be delayed by up to min(miss_timeout / 2, megamiss_timeout / 2, kill_timeout / 2, multikill_timeout / 2).

Reply:
changed to last_checkin_ for consistency after your recent change